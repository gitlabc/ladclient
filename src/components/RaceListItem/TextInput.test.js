import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './TextInput';
import {shallow} from 'enzyme';

describe('TextInput', () => {
    
    test('should toggles erro type when show/hide password clicked', () => {
        const wrapper1 = shallow(<TextInput 
            htmlId="example-optional"
            label="First Name"
            name="firstname"
            onChange={() => { }}
            required
        />);

        expect(wrapper1.find('div')).toHaveLength(1);
        
        const wrapper2 = shallow(<TextInput 
            htmlId="example-optional"
            label="First Name"
            name="firstname"
            onChange={() => { }}
            required
            error="First name is required."            
        />);

        expect(wrapper2.find('div')).toHaveLength(2);
    });
    

    test('should hides error by default', () => {
        const tree = renderer.create(
            <TextInput
                htmlId="example-optional"
                label="First Name"
                name="firstname"
                onChange={() => { }}
                required
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('should shows error when error is set', () => {
        const tree = renderer.create(
            <TextInput
                htmlId="example-optional"
                label="First Name"
                name="firstname"
                onChange={() => { }}
                required
                error="First name is required."
            />
        ).toJSON(); 

        expect(tree).toMatchSnapshot();
    });
    
});
