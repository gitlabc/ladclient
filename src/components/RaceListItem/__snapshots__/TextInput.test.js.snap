// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should hides error by default 1`] = `
<div
  className={undefined}
>
  <label
    htmlFor="example-optional"
    style={
      Object {
        "display": "block",
      }
    }
  >
    First Name
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </label>
  <input
    className={undefined}
    id="example-optional"
    name="firstname"
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value={undefined}
  />
</div>
`;

exports[`TextInput should shows error when error is set 1`] = `
<div
  className={undefined}
>
  <label
    htmlFor="example-optional"
    style={
      Object {
        "display": "block",
      }
    }
  >
    First Name
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </label>
  <input
    className={undefined}
    id="example-optional"
    name="firstname"
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value={undefined}
  />
  <div
    className={undefined}
  >
    First name is required.
  </div>
</div>
`;
